test/components/embedders/types/test_protocol.py

```diff
@@ -0,0 +1,36 @@
+from typing import Any, Dict
+
+from haystack.components.embedders.types.protocol import TextEmbedder
+
+
+class MockTextEmbedder:
+    def run(self, text: str, param_a: str = "default", param_b: str = "another_default") -> Dict[str, Any]:
+        """
+        Mock implementation that returns a simple embedding and metadata.
+        """
+        return {"embedding": [0.1, 0.2, 0.3], "metadata": {"text": text, "param_a": param_a, "param_b": param_b}}
+
+
+def test_protocol_implementation():
+    embedder: TextEmbedder = MockTextEmbedder()  # should not raise any type errors
+
```

Comment:
In addition, we can check here the signature of the embedders run method with something like:

```python
    import inspect
    run_signature = inspect.signature(MockTextEmbedder.run)
    assert "text" in run_signature.parameters
    assert run_signature.parameters["text"].annotation == str
    assert run_signature.return_annotation == Dict[str, Any] 
```
or the last line could use EmbeddingOutput if we add it
```python
    assert run_signature.return_annotation == EmbeddingOutput 
```